lib/rspec/rails/matchers/have_http_status.rb

```diff
@@ -18,7 +18,12 @@ module HaveHttpStatus
         # @return response matcher instance
         def self.matcher_for_status(target)
           if GenericStatus.valid_statuses.include?(target)
```

Comment:
Rather than check valid statuses and then doing the mapping, this could do the mapping, then check valid statuses. `status_method = GenericStatus.map_status(target); if (status_method) ...` Then GenericStatus encapsulates all of the logic. I'm willing to make that change.

Reply:
I'd still like to see this class done away with, and the lookup method defined depending on the version of rails. See below.